### ISSUE TITLE
`x.real[:] = y` and `x.imag[:] = y` do not mutate x

### ISSUE THREAD

**Issue by kohr-h** [cupy/cupy#799]
Since `ndarray.real` is implemented as a ufunc and not as a view, `x.real` does not share memory with `x` and will not mutate `x` upon assignment. Hence, `x.real = y` and `x.real[:] = y` behave differently.

Wouldn't it be possible to make a view and manipulate `strides` and `itemsize`? Like this:
```python
    @property
    def real(self):
        if self.dtype.kind == 'c':
            self_view = self.view(self.dtype.char.lower())
            self_view._set_shape_and_strides(self.shape, self.strides)
            return self_view
        else:
            return self
```
 For `imag` it seems harder since CUDA apparently needs memory addresses to be aligned at multiples of the `itemsize`. Is there a workaround for that?

**Comment by kohr-h** [comment]
If not, maybe a mention in the docs and the code would be good then.

**Comment by kmaehashi** [comment]
Thank you for raising an issue and suggestion, I started to work on this in #1089.

**Comment by kmaehashi** [comment]
Fixed via #1089.
